### PR TITLE
Elastic agent on k8s

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
Add manifests and docs regarding running Agent on k8s.

## Why is it important?
It's a suggested way to run Agent on k8s environments in oder to deploy Kubernetes Package (https://github.com/elastic/integrations/pull/70) efficiently.

cc: @exekias

### CODE REVIEW

File: x-pack/elastic-agent/docs/running-on-kubernetes.asciidoc [hunk 1]
```diff
@@ -0,0 +1,109 @@
+[[running-on-kubernetes]]
+[role="xpack"]
+=== Run {agent} on Kubernetes
+
+You can use {agent} https://www.docker.elastic.co/r/beats/elastic-agent[Docker images] on Kubernetes to
+retrieve cluster metrics.
+
+ifeval::["{release-state}"=="unreleased"]
+
+However, version {version} of {agent} has not yet been
```
Comment (sayden) [inline x-pack/elastic-agent/docs/running-on-kubernetes.asciidoc:10]: I'd delete this paragraph or we may risk to forget about it in the near future. In other words, docs shouldn't "warn" or "announce" future updates or changes

Comment (ChrsMark) [reply x-pack/elastic-agent/docs/running-on-kubernetes.asciidoc:10]: I think we are covered by the `ifeval::["{release-state}"=="unreleased"]` :)
The same we do also in manifests for x-Beats too: https://github.com/elastic/beats/blame/master/metricbeat/docs/running-on-kubernetes.asciidoc#L9

**Comment by ruflin** [comment]
Note: The points below should not block this PR.

@exekias Is this just a "recommendation" or our "supported" way or running the agent on k8s?
@ph We should probably also have a follow up discussion on who owns updating these manifests?

File: deploy/kubernetes/elastic-agent-kubernetes.yaml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    spec:
+      serviceAccountName: agent-ingest-management
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      containers:
+        - name: agent-ingest-management-nodescope
+          image: docker.elastic.co/beats/elastic-agent:8.0.0-SNAPSHOT
+          command: ["/bin/sh", "-c"]
+          args: [
+            "yes | ./elastic-agent enroll $KIBANA_HOST $FLEET_ENROLLMENT_TOKEN && ./elastic-agent run"
```
Comment (exekias) [inline deploy/kubernetes/elastic-agent-kubernetes.yaml:28]: uhm I wonder if this should be handled by the agent image itself. What will happen here when we are already enrolled but get a pod recreation?

**Comment by exekias** [comment]
> Note: The points below should not block this PR.
> 
> @exekias Is this just a "recommendation" or our "supported" way or running the agent on k8s?

This should be the supported way of running the agent, in many cases you can expect some user tweaks. In any case, as of now It feels more like a first stab at it, we should keep iterating over this story and make it sound for GA

**Comment by ChrsMark** [comment]
> I saw the last change in the run command, let's discuss that before merging

This change performs the same commands with what `docker-entrypoint` performs: https://github.com/elastic/beats/blob/master/dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl
1. enroll: https://github.com/elastic/beats/blob/7579d512bfcf537b1f259b01769e3abc718ad58e/dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl#L56
2. start: https://github.com/elastic/beats/blob/7579d512bfcf537b1f259b01769e3abc718ad58e/dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl#L62

This was used to work with master versions because of https://github.com/elastic/beats/pull/18445/files#diff-4847a986cf06ab4d26f26f5212720eedR52. However the respective PR was not backported to `7.9` (see https://github.com/elastic/beats/blob/ab4198642341c1b909e2a30f01cd0808aa786a9c/dev-tools/packaging/templates/docker/docker-entrypoint.elastic-agent.tmpl#L53) and hence the startup script was failing cause the `apikey` was always overwritten.

I agree we should not patch the startup command of the container with custom one. I will open a PR with this small change for  `7.9`.

**Comment by ChrsMark** [comment]
Tested the manifests e2e with a fresh 7.9 docker image. Previous issue resolved, merging this one.

**Pull request merged**